### ISSUE TITLE
Unable to reploy deployment

### ISSUE THREAD

**Issue by dbd** [rancher/dashboard#1765]
When attempting to redeploy a deployment I see the following in the browser dev tools:

```json
{
    "type": "error",
    "links": {},
    "code": "BadRequest",
    "message": "Deployment in version \"v1\" cannot be handled as a Deployment: v1.Deployment.Spec: v1.DeploymentSpec.Template: v1.PodTemplateSpec.Spec: v1.PodSpec.Affinity: v1.Affinity.NodeAffinity: v1.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution: v1.NodeSelector.NodeSelectorTerms: []v1.NodeSelectorTerm: v1.NodeSelectorTerm.MatchExpressions: []v1.NodeSelectorRequirement: v1.NodeSelectorRequirement.Values: []string: decode slice: expect [ or n, but found \", error found in #10 byte of ...|\"values\":\"true\"}]}]}|..., bigger context ...|ssions\":[{\"key\":\"iscsi\",\"operator\":\"In\",\"values\":\"true\"}]}]}}},\"containers\":[{\"env\":[{\"_display\":{\"b|...",
    "status": 400
}

```

The relevent part of deployment.yaml
```yaml
    spec:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: iscsi
                operator: In
                values:
                - "true"
```

**Comment by richard-cox** [comment]
The product has moved on from a few years ago and in 2.7-head (2.7.0 timeframe) I couldn't reproduce this with a Deployment (using `ubuntu` container image) and the following yaml

```
spec:
  template:
    spec:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: node-role.kubernetes.io/master
                operator: In
                values:
                - "true"
```

Bumping this to QA in case there was any angle that i missed

**Comment by jameson-mcghee** [comment]
During testing on `v2.7-head` (Commit ID: ) I redeployed the `rancher/rancher:v2.7-head` Deployment on the `local` cluster and then redeployed all the Deployments on the `local` cluster. During testing, I did not observe the reported error in the Network Logs. Therefore I am closing this ticket.

[jrm27headrancher121422 Network Logs.zip](https://github.com/rancher/dashboard/files/10238764/jrm27headrancher121422.Network.Logs.zip)
